Source/Core/Core/IOS/Network/SSL.cpp

```diff
@@ -35,6 +35,40 @@ static constexpr mbedtls_x509_crt_profile mbedtls_x509_crt_profile_wii = {
     0,         /* No RSA min key size */
 };
 
+namespace
+{
+// Dirty workaround to disable SNI which isn't supported by the Wii.
+//
+// This SSL extension can ONLY be disabled by undefining
+// MBEDTLS_SSL_SERVER_NAME_INDICATION and recompiling the library. When
+// enabled and if the hostname is set, it uses the SNI extension which is sent
+// with the Client Hello message.
+//
+// This workaround doesn't require recompiling the library. It does so by
+// deferring mbedtls_ssl_set_hostname after the Client Hello message. The send
+// callback is used as it's the (only?) hook called at the beginning of
+// each step of the handshake by the mbedtls_ssl_flush_output function.
+//
+// The hostname still needs to be set as it is checked against the Common Name
+// field during the certificate verification process.
+int SSLSendWithoutSNI(void* ctx, const unsigned char* buf, size_t len)
+{
+  auto* mbed_ssl_ctx = static_cast<mbedtls_ssl_context*>(ctx);
+
+  if (mbed_ssl_ctx->state == MBEDTLS_SSL_SERVER_HELLO)
+  {
+    const auto& ssl_table = IOS::HLE::Device::NetSSL::_SSL;
+    auto it = std::find_if(std::begin(ssl_table), std::end(ssl_table),
+                           [ctx](const IOS::HLE::WII_SSL& ssl) { return &ssl.ctx == ctx; });
+    if (it != std::end(ssl_table))
+      mbedtls_ssl_set_hostname(mbed_ssl_ctx, it->hostname.c_str());
+    else
+      ERROR_LOG(IOS_SSL, "Failed to apply SNI hostname workaround!");
+  }
+  return mbedtls_net_send(ctx, buf, len);
```

Comment:
This is incorrect. mbedtls_net_send should be passed `&ssl->hostfd` (the socket), not the SSL context.